ShortQuestions/hw1.md

```diff
@@ -0,0 +1,82 @@
+### 1. Learn MarkDown and show all of basic usage in the ShortQuestions/README.md
+
+SEE README.md
+
+### 2. practice git using the platform. list the git commands you learned
+
+```
+git status
+git log
+git commit
+git add
+git pull
+git push
+git merge
+git rebase
+git fetch
+```
+
+### 3. What is the basic steps to init a git repo in you local?
+
+```bash
+git init
+```
+
+### 4. How to clone a repo from Github?
+
+```bash
+git clone repo_url
+```
+
+### 5. How to create a new branch and checkout to that branch?
+
+```bash
+git cehckout -b branch_name
+```
+
+### 6. How to merge the branch_test to master branch in command? show me the commands.
+
+```bash
+git checkout master
+git pull origin master
+git merge branch_test
+git push origin master
+```
+
+### 7. How to stash your new code before leaving branch branch_learn_stash and pop your stash when you checkout back to \*\*branch_learn_stash ? try commands way and intellij way.
+
+```bash
+git checkout branch_learn_stash
+git stash
+git checkout different_branch
+git checkout branch_learn_stash
+git stash pop
+```
+
+### 8. How do you understand PR is based on Branch?
+
+- A Pull Request (PR) allows you to merge changes from one branch to the main codebase, while also enabling a review process by other developers.
+
+### 9. What is maven role? what it be used to do?
+
+- Maven helps automate the build process and manage project dependencies for Java projects. It is similar to `apt` in Linux and `npm` in Node.js.
```

Comment:
Well done! But don't think apt is an appropriate example here. Because apt is for *system-level* software management for linux while maven and npm are *project-level* tools for different languages.

Reply:
Thank you! You’re right. Apt is indeed more of a system-level package manager for Linux, whereas Maven and npm are project-level tools specific to Java and JavaScript respectively. I appreciate the clarification!